codegen/src/getCli.ts

```diff
@@ -0,0 +1,70 @@
+import { unlink, writeFile } from 'fs/promises';
+import fetch from 'node-fetch';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import tar from 'tar';
+import unzipper from 'unzipper';
+import { Ora } from 'ora';
+import { createReadStream, createWriteStream } from 'fs';
+
+import { cliPath } from './cliPath';
+import chalk from 'chalk';
+import { getCliPlatformFromNodePlatform } from './getCliPlatformFromNodePlatform';
+
+type GitHubResponse = {
+  assets: { browser_download_url: string }[];
+};
+
+export const getCli = async ({ spinner }: { spinner: Ora }) => {
+  console.log();
+  spinner.info(
+    `You don't have the Xata CLI installed on your system, so we're downloading it. You could make this process faster by installing the CLI locally. More info: ${chalk.blueBright(
+      'https://docs.xata.io/cli/getting-started'
+    )}.
+`
+  );
+  spinner.start('Looking up latest Xata CLI...');
+  const fileUrl = await fetch('https://api.github.com/repos/xataio/cli/releases/latest')
+    .then((r) => r.json())
+    .then((d: GitHubResponse) =>
+      d.assets.map((a) => a.browser_download_url).find((a) => a.includes(getCliPlatformFromNodePlatform()))
+    );
+
+  if (!fileUrl) {
+    failWithIncompatibleOs({ spinner });
```

Comment:
We should print a different message when the list of download urls is empty. This can happen if a release is not generated correctly or their files are being uploaded and are not ready.